code/game/objects/items/hot_potato.dm

```diff
@@ -0,0 +1,137 @@
+	icon = 'icons/obj/hydroponics/harvest.dmi'
+	icon_state = "potato"
+	flags_1 = NOBLUDGEON_1
+	force = 0
+	var/icon_off = "potato"
+	var/icon_on = "potato_active"
+	var/detonation_timerid
+	var/activation_time = 0
+	var/timer = 600			//deciseconds
+	var/show_timer = FALSE
+	var/reusable = FALSE		//absolute madman
+	var/sticky = TRUE
+	var/forceful_attachment = TRUE
+	var/stimulant = TRUE
+	var/detonate_explosion = TRUE
+	var/detonate_dev_range = 0
+	var/detonate_heavy_range = 0
+	var/detonate_light_range = 2
+	var/detonate_flash_range = 5
+	var/detonate_fire_range = 5
+
+/obj/item/hot_potato/proc/detonate()
+	if(detonate_explosion)
+		explosion(src, detonate_dev_range, detonate_heavy_range, detonate_light_range, detonate_flash_range, flame_range = detonate_fire_range)
+	deactivate()
+	if(!reusable)
+		var/mob/M = loc
+		if(istype(M))
+			M.dropItemToGround(src, TRUE)
+		qdel(src)
+
+/obj/item/hot_potato/proc/is_active()
+	return isnull(detonation_timerid)? FALSE : TRUE
+
+/obj/item/hot_potato/attack_self(mob/user)
+	if(activate(timer, user))
+		user.visible_message("<span class='boldwarning'>[user] squeezes [src], which promptly starts to flash red-hot colors!</span>", "<span class='boldwarning'>You squeeze [src], activating its countdown and attachment mechanism!</span>",
+		"<span class='boldwarning'>You hear a mechanical click and a loud beeping!</span>")
+		return
+	return ..()
+
+/obj/item/hot_potato/process()
+	if(stimulant)
+		if(isliving(loc))
+			var/mob/living/L = loc
+			L.SetStun(0)
+			L.SetKnockdown(0)
+			L.SetSleeping(0)
+			L.SetUnconscious(0)
+			L.reagents.add_reagent("muscle_stimulant", CLAMP(5 - L.reagents.get_reagent_amount("muscle_stimulant"), 0, 5))	//If you don't have legs or get bola'd, tough luck!
+
+/obj/item/hot_potato/examine(mob/user)
+	. = ..()
+	if(is_active())
+		to_chat(user, "<span class='warning'>[src] is flashing red-hot! You should probably get rid of it!</span>")
+		if(show_timer)
+			to_chat(user, "<span class='warning'>[src]'s timer looks to be at [(activation_time - world.time) / 10] seconds!</span>")
+
+/obj/item/hot_potato/equipped(mob/user)
+	. = ..()
+	if(is_active())
+		to_chat(user, "<span class='userdanger'>You have a really bad feeling about [src]!</span>")
+
+/obj/item/hot_potato/afterattack(atom/target, mob/user, adjacent, params)
+	if(!adjacent || !ismob(target))
+		return ..()
+	force_onto(target, user)
+
+/obj/item/hot_potato/proc/force_onto(mob/living/victim, mob/user)
+	if(!istype(victim) || user != loc || victim == user)
+		return FALSE
+	if(!victim.client)
+		to_chat(user, "<span class='boldwarning'>[src] refuses to attach to a non-sapient creature!</span>")
+	if(victim.stat != CONSCIOUS || !victim.get_num_legs())
+		to_chat(user, "<span class='boldwarning'>[src] refuses to attach to someone incapable of using it!</span>")
+	user.dropItemToGround(src, TRUE)

```

Comment:
you should be using temporarilyRemoveItemFromInventory. Just put it back if you can't move it